File: backend/src/main/java/com/pofo/backend/domain/board/dto/BoardResponseDto.java
package com.pofo.backend.domain.board.dto;

import com.pofo.backend.domain.board.entity.Board;
import lombok.Getter;
import java.time.LocalDateTime;



//게시글 상세 조회 -> GET /api/v1/user/boards/{id}

@Getter
public class BoardResponseDto  {
    private final Long id;  //게시글 ID
    private final String title;
    private final String content;
    private final LocalDateTime createdAt;  // 작성일자 필드 추가
    private final LocalDateTime updatedAt;  // 수정일자 필드 추가
    private Long userId;  // ✅ 작성자 ID 추가



    public BoardResponseDto(Board board) {
        this.id = board.getId();
        this.title = board.getTitle();
        this.content = board.getContent();
        this.createdAt = board.getCreatedAt();  // BaseTime의 필드 값 설정
        this.updatedAt = board.getUpdatedAt();  // 수정일자 추가
        this.userId = board.getUser().getId();  // ✅ 작성자의 userId 포함

    }
}

File: backend/src/main/java/com/pofo/backend/domain/board/dto/BoardResponseDto.kt

package com.pofo.backend.domain.board.dto

import com.pofo.backend.domain.board.entity.Board
import java.time.LocalDateTime


data class BoardResponseDto(
    val id: Long?,  // 게시글 ID
    val title: String,
    val content: String,
    val createdAt: LocalDateTime?,  // 작성일자 필드 추가
    val updatedAt: LocalDateTime?,  // 수정일자 필드 추가
    val userId: Long?  // 작성자 ID 추가
) {
    constructor(board: Board) : this(
        id = board.id,
        title = board.title,
        content = board.content,
        createdAt = board.createdAt,
        updatedAt = board.updatedAt,
        userId = board.user.id  // :흰색_확인_표시: Reflection 없이 직접 접근
    )
}



// Base엔티티 @Getter적용, board.id 대신 board.getId() 사용, private필드에 접근불가
// createdAt, updatedAt도 board.getCreatedAt() 사용